src/main/java/org/terasology/dynamicCities/facets/Grid2DFacet.java

```diff
@@ -35,18 +36,18 @@
     protected Rect2i gridWorldRegion;
     protected Rect2i gridRelativeRegion;
 
-    public Grid2DFacet(Region3i targetRegion, Border3D border, int gridSize) {
+    public Grid2DFacet(BlockRegion targetRegion, Border3D border, int gridSize) {
         super(targetRegion, border);
         this.gridSize = gridSize;
-        center = new Vector2i(targetRegion.center().x(), targetRegion.center().z());
-        gridWorldRegion = Rect2i.createFromMinAndMax(center.x() - targetRegion.sizeX() / (2 * gridSize),
-                center.y() - targetRegion.sizeY() / (2 * gridSize),
-                center.x() + targetRegion.sizeX() / (2 * gridSize),
-                center.y() + targetRegion.sizeY() / (2 * gridSize));
+        center = new Vector2i(targetRegion.center(new Vector3f()).x(), targetRegion.center(new Vector3f()).z());
```

Comment:
This allocates two vectors for the same thing - let's use a local variable and only allocate one vector.
```suggestion
		Vector3f regionCenter = targetRegion.center(new Vector3f());
        center = new Vector2i(regionCenter.x(), regionCenter.z());
```